### PR TITLE
Add npm-init and Hapi generator

### PR DESCRIPTION
**Pull request by chribben**

### CODE REVIEW

File: generators/hapi/index.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  install: {
+
+    installDependencies: function() {
+      const dependencies = [
+        'hapi',
+        'hapi-auth-basic',
+        'swaggerize-hapi',
+        '@springworks/hapi-default-extensions',
+        '@springworks/logger-factory',
+        '@springworks/bootstrapper',
```
Comment (petrenkotino) [inline generators/hapi/index.js:57]: how about `devDependencies`, ie. `swagger-md` and `swagger-tools`? Maybe another generator adds them?

File: generators/npm-init/index.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    let guessed_name = path.basename(this.destinationRoot());
+    if (guessed_name.substring(0, 3) === 'm2h-') {
+      guessed_name = guessed_name.substring(4, guessed_name.length - 1);
+    }
+
+    const prompts = [
+      {
+        type: 'input',
+        name: 'server_name',
+        message: 'Name of the server (full name will be @springworks/{server_name})',
```
Comment (petrenkotino) [inline generators/npm-init/index.js]: We usually annotate private npm modules with `@springworks`.

File: generators/npm-init/templates/_package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "@springworks/<%=server_name%>",
```
Comment (petrenkotino) [inline generators/npm-init/templates/_package.json]: We usually annotate private npm modules with `@springworks`.

**Review by petrenkotino — APPROVED** [review]

File: generators/hapi/index.js (hunk 1 above)
Comment (chribben) [reply generators/hapi/index.js:57]: Yeah, `npm-init` adds them

**Pull request merged**